src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -570,7 +570,17 @@ def save_as_table(
         child: SnowflakePlan,
     ) -> SnowflakePlan:
         full_table_name = ".".join(table_name)
+
+        # here get the column definition from the child attributes. In certain cases we have
+        # the attributes set to ($1, VariantType()) which cannot be used as valid column name
+        # in save as table. So we rename ${number} with COL{number}.
+        hidden_column_pattern = r"\$(\d+)"
         column_definition = attribute_to_schema_string(child.attributes)
+        column_definition = re.sub(
```

Comment:
NIT: Can we combine the two assignments?

Reply:
I was thinking about joining it too but I decided against it to make code more readable. We build column definition in two steps. 1. converting attributes to column_definition and 2 is rename hidden column patterns into valid ones.
I am still okay with changing into a single statement.

Reply:
If we want to keep them separate, should we consider giving the unprocessed variable a different name for readability, for example, `column_definition_with_hidden_columns`?

Reply:
I like that